pkg/controller/summon/components/rotate_fernet.go

```
@@ -0,0 +1,129 @@
+/*
+Copyright 2018 Ridecell, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package components
+
+// TODO: This whole thing should probably be its own custom resource.
+
+import (
+	"crypto/rand"
+	"encoding/base64"
+	"fmt"
+	"time"
+
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	summonv1beta1 "github.com/Ridecell/ridecell-operator/pkg/apis/summon/v1beta1"
+	"github.com/Ridecell/ridecell-operator/pkg/components"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+)
+
+const customTimeLayout = "2006-01-02T15-04-05"
+
+type fernetRotateComponent struct{}
+
+func NewFernetRotate() *fernetRotateComponent {
+	return &fernetRotateComponent{}
+}
+
+func (comp *fernetRotateComponent) WatchTypes() []runtime.Object {
+	return []runtime.Object{
+		&corev1.Secret{},
+	}
+}
+
+func (_ *fernetRotateComponent) IsReconcilable(ctx *components.ComponentContext) bool {
+	instance := ctx.Top.(*summonv1beta1.SummonPlatform)
+
+	fernetSecret := &corev1.Secret{}
+	err := ctx.Get(ctx.Context, types.NamespacedName{Name: fmt.Sprintf("%s.fernet-keys", instance.Name), Namespace: instance.Namespace}, fernetSecret)
+	// If secret doesn't exist reconcile to create new
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			return true
+		}
+		return false
```

This would mean that an error getting the secret would be silently dropped, which is not good.